### PR TITLE
Show location description in draft/edit component list item

### PR DESCRIPTION
**Pull request by roseeichelmann**
## Associated issues
Closes https://github.com/cityofaustin/atd-data-tech/issues/14166

## Testing
**URL to test:** 
https://deploy-preview-1156--atd-moped-main.netlify.app/moped/projects/1759?tab=activity_log

**Steps to test:**
1. Go to a projects map view.
2. Create a new component with a location description. When mapping the component in create mode, the component list item should have the Component Type - Subtype on the first line, and the second line should have the location description, similar to the non-editing state of a components list item.
image
4. Save the component and test editing it. The above should also be true.

---
#### Ship list
- [ ] Code reviewed 
- [ ] Product manager approved
- [ ] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

File: moped-editor/src/views/projects/projectView/ProjectComponents/DraftComponentListItem.js
Comment (mddilley) [inline moped-editor/src/views/projects/projectView/ProjectComponents/DraftComponentListItem.js:1]: The bug:
1. Create a new component with a location description, map it, and save
2. Edit the same component's location description and save
3. Edit the same component's features using the map and you will see the outdated location description

This is due to the state updates being scattered and hard to grok, but @johnclary  centralized those updates in https://github.com/cityofaustin/atd-moped/pull/1150, and I tested this branch with that branch merged in and the bug is gone. 🪄

**Review by mddilley — COMMENTED** [review]
Looks great and I love that this makes use of an existing hooks to have less props. 🙌 I found a small bug, but I think it is an existing one that will be fixed by another open PR. Might just mean that we make sure they ship together. Ready to approve when I get some back up on that.

File: moped-editor/src/views/projects/projectView/ProjectComponents/DraftComponentListItem.js
Comment (johnclary) [reply moped-editor/src/views/projects/projectView/ProjectComponents/DraftComponentListItem.js:1]: nice—a freebie! 🙌

File: moped-editor/src/views/projects/projectView/ProjectComponents/DraftComponentListItem.js [hunk 1]
```diff
@@ -6,15 +6,17 @@ import ListItemText from "@mui/material/ListItemText";
...
 const DraftComponentListItem = ({
-  primaryText,
-  secondaryText,
+  component,
   onSave,
   onCancel,
   saveButtonText,
   saveButtonDisabled,
 }) => {
+  const { primary, secondary } = useComponentListItemText(component);
```
Comment (johnclary) [inline moped-editor/src/views/projects/projectView/ProjectComponents/DraftComponentListItem.js:18]: yay for having a hook ready to handle this.

i just had to code this out in a SQL view for the AGOL integration, and now it occurs to me that we should probably be using a generated column to create the `<name> - <subtype>` string at the DB level. not a big deal and i'm glad to see this hook reused 🚀

**Review by johnclary — APPROVED** [review]
looks great and works great! thanks for helping us giving Moped some polish! 🙌 🚢 🚢 🚢 🚢 🚢 🚢 🚢 🚢

**Review by mddilley — APPROVED** [review]
Yay! So great to see more component data visible in these flows. ✨

**Pull request merged**